### PR TITLE
test: make equalNode matcher resilient to null actual

### PR DESCRIPTION
**Pull request by pavan-sh**
The equalNode custom matcher could throw when the received value was null/undefined because it accessed obj.tagName before checking obj.
This makes it return a normal failed assertion with a helpful message instead of crashing, improving test diagnostics and robustness across environments.

Note: **LLM Assisted PR**

### CODE REVIEW

**Comment by pavan-sh** [comment]
> Can you clarify when/where this is useful?

Sure, this helps when a test expects a DOM node but the actual value ends up being null/undefined (e.g. a querySelector didn’t find an element, or document.activeElement isn’t what we expect).

Before this change, equalNode could throw while building the pass/fail result because it accessed `obj.tagName` even when `obj` was null, which hides the real assertion failure and makes debugging harder. With this change, the matcher fails cleanly and prints a readable “expected node X, received null/undefined” message instead.

**Comment by rschristian** [comment]
> when a test expects a DOM node but the actual value ends up being null/undefined (e.g. a querySelector didn’t find an element, or document.activeElement isn’t what we expect).

And where is that occurring, is my question? Seems like a non-issue, though I don't have anything against the change necessarily.

**Comment by pavan-sh** [comment]
> > when a test expects a DOM node but the actual value ends up being null/undefined (e.g. a querySelector didn’t find an element, or document.activeElement isn’t what we expect).
> 
> And where is that occurring, is my question? Seems like a non-issue, though I don't have anything against the change necessarily.

Fair question, I’m not claiming there’s an existing failing test right now. This is a defensive improvement to the matcher itself.

The current implementation can throw if `equalNode` is used in a test where the received value is unexpectedly null/undefined (e.g. `document.querySelector(...)` didn’t find an element, or `document.activeElement` is null in a headless/browser edge case). In that situation the matcher throws while evaluating `pass` (accessing `obj.tagName`), which hides the original assertion context.

With this change, those cases produce a normal assertion failure with a clear message instead of crashing the matcher.
If you prefer, I can also add a tiny unit test for `equalNode` covering the null/undefined case to demonstrate the behavior.

File: vitest.setup.js [hunk 1]
```diff
@@ -31,12 +31,20 @@ expect.extend({
...
+
+		return {
+			pass,
+			message: () => {
+				if (obj == null) {
+					return `expected node to have tagName ${expected.tagName} but got ${obj} instead.`;
+				}
+
+				return `expected node to have tagName ${expected.tagName} but got ${obj.tagName} instead.`;
+			}
```
Comment (rschristian) [inline vitest.setup.js]:
```suggestion
			message: () =>
				`expected node to have tagName ${expected.tagName} but got ${obj ? obj.tagName : obj} instead.`;
```

Comment (pavan-sh) [reply vitest.setup.js]: Sorry, you’re right to call this out.

I wasn’t aware disclosure was explicitly expected here. I did use an AI assistant while drafting parts of the PR/replies, and I should have mentioned it. I’ve read the Code of Conduct and I’ll disclose it going forward.

I also updated the PR to follow your suggested minimal fix: the matcher message now guards `obj.tagName` (`${obj ? obj.tagName : obj}`) so it won’t throw when `obj` is null/undefined.

Thanks for the feedback and your valuable time.

Comment (rschristian) [reply vitest.setup.js]: > I also updated the PR to follow your suggested minimal fix: the matcher message now guards `obj.tagName` (`${obj ? obj.tagName : obj}`) so it won’t throw when `obj` is null/undefined.

*sigh* No, my suggestion had absolutely nothing to do with guarding `obj.tagName`, you already had that down just fine with the null check. My suggestion was just to reduce duplication, it fixed nothing.

I don't think you actually understand the code that you're suggesting here.

**Comment by rschristian** [comment]
You didn't read the Code of Conduct, you're not reading or understanding the changes you're proposing, my replies are clearly just being passed directly to your LLM, and this is of questionable value at best. It's a dead PR, I don't think we should entertain this any further.

I'm closing this out.
